R/MappedDataTimeProfile.R

```diff
@@ -0,0 +1,325 @@
+    #' @param y2limits limits for secondary axis (may be NULL)
+    #'
+    #' @description Create a new `MappedDataTimeProfile` object
+    #' @return A new `MappedDataTimeProfile` object
+    initialize = function(data,
+                          mapping,
+                          groupAesthetics = NULL,
+                          groupOrder = NULL,
+                          direction = "y",
+                          isObserved = TRUE,
+                          xlimits = NULL,
+                          ylimits = NULL,
+                          scaleOfPrimaryAxis = "linear",
+                          scaleOfSecondaryAxis = "linear",
+                          y2limits = NULL) {
+      super$initialize(
+        data = data,
+        mapping = mapping,
+        groupAesthetics = groupAesthetics,
+        groupOrder = groupOrder,
+        direction = direction,
+        isObserved = isObserved,
+        xlimits = NULL,
+        ylimits = NULL
+      )
+
+      checkmate::assertChoice(scaleOfPrimaryAxis, choices = c("linear", "log"))
+      private$scaleOfPrimaryAxis <- scaleOfPrimaryAxis
+
+      # check for secondaryAxis
+      if ("y2axis" %in% names(self$mapping)) {
+        checkmate::assertLogical(private$getDataForAesthetic(aesthetic = "y2axis"),
+          any.missing = FALSE, .var.name = "y2axis",
+        )
+
+        private$secondaryAxisAvailable <- any(private$getDataForAesthetic("y2axis"))
+
+        private$addOverwriteAes(newMaps = aes(y2 = y2))
+      } else {
+        private$secondaryAxisAvailable <- FALSE
+      }
+
+
+      # set fields for secondary Axis
+      if (private$secondaryAxisAvailable) {
+        checkmate::assertChoice(scaleOfSecondaryAxis,
+          choices = c("linear", "log"),
+          null.ok = TRUE
+        )
+        checkmate::assertDouble(
+          y2limits,
+          sorted = TRUE,
+          any.missing = TRUE,
+          len = 2,
+          unique = TRUE,
+          null.ok = TRUE
+        )
+
+
+        private$scaleOfSecondaryAxis <- scaleOfSecondaryAxis
+        self$ylimits <- ylimits
+        self$y2limits <- y2limits
+
+        # setLimits
+        private$setyLimits()
+
+        return(invisible(self))
+      }
+    },
+    #' scales data for secondary axis and updates filed secAxis
+    #'
+    #' @param ylimits limits for primary axis
+    #' @param y2limits limits for secondary axis
+    #' @param y2scale.args arguments for secondary axis
+    #'
+    #' @return updated MappedDataTimeProfile
+    scaleDataForSecondaryAxis = function(ylimits = NULL,
+                                         y2limits = NULL,
+                                         y2scale.args = list()) {
+      if (!self$requireDualAxis) {
+        private$dataScaled <- self$data
+        return(invisible(self))
+      }
+
+      checkmate::assertDouble(
+        ylimits,
+        sorted = TRUE,
+        any.missing = TRUE,
+        len = 2,
+        unique = TRUE,
+        null.ok = TRUE
+      )
```

Comment:
I have seen this function several times, and all arguments (except the first one) are the same.
Consider introducing the function with just 1 argument ("checkLimits(limits)" or whatsoever)